Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml

```diff
@@ -191,6 +191,58 @@
         <UnloadPeriod value="0.5"/>
         <LoadValidatorArray value="NotSpawnling"/>
     </CAbilTransport>
+    <CAbilTransport id="AP_TradeStructureTransport">
+        <EditorCategories value="Race:Zerg,AbilityorEffectType:Units"/>
+        <CmdButtonArray index="Load" DefaultButtonFace="AP_TradeStructureLoad" State="Restricted" Requirements="AP_NotTradeStructureBusy"/>
+        <CmdButtonArray index="UnloadAll" DefaultButtonFace="AP_TradeStructureUnloadAll" State="Restricted" Requirements="AP_NotTradeStructureBusy">
+            <Flags index="ToSelection" value="1"/>
+        </CmdButtonArray>
+        <CmdButtonArray index="UnloadAt">
+            <Flags index="Hidden" value="1"/>
+        </CmdButtonArray>
+        <CmdButtonArray index="UnloadUnit">
+            <Flags index="Hidden" value="1"/>
+        </CmdButtonArray>
+        <CmdButtonArray index="LoadAll">
+            <Flags index="Hidden" value="1"/>
+        </CmdButtonArray>
+        <Flags index="CargoDeath" value="1"/>
+        <Flags index="ShowCargoSize" value="0"/>
+        <Flags index="ShowStatus" value="0"/>
+        <Range value="0.5"/>
+        <MaxCargoCount value="255"/>
+        <MaxCargoSize value="8"/>
+        <TotalCargoSpace value="1020"/>
+        <InitialUnloadDelay value="0.5"/>
+        <LoadPeriod value="0.25"/>
+        <UnloadPeriod value="0.5"/>
+        <TargetFilters value="Visible;Ally,Neutral,Enemy,Heroic,Buried,Dead,Hidden,Hallucination,Summoned,HeroUnit"/>
+        <LoadValidatorArray value="HasNoCargo"/>
+        <LoadValidatorArray value="AP_NotIsMindControlled"/>
+        <LoadValidatorArray value="AP_NotHasTimedLife"/>
+        <!-- It's possible to MC a vanilla Zergling and morph it, removing the MC behavior -->
+        <LoadValidatorArray value="AP_IsNotVanillaBaneling"/>

```

Comment:
Check enemy Corruptors MC'd in All-In and morph'd to Brood Lords

Reply:
Tested: Morphed Brood Lords actually keep the MC behavior, so Banelings losing it on morph might be a vanilla bug?
![SC2_x64_dIfpmVHXxs](https://github.com/user-attachments/assets/87503a64-dfb9-4b0f-96bf-4a2cf9d406c1)

Of course, both the MC behavior and them having 0 cargo size from being a vanilla air unit prevent them from being tradable.